pkg/tektonlog/tektonlog.go

```diff
@@ -211,91 +211,79 @@ func (t *TektonLogger) GetRunningBuildLogs(ctx context.Context, pa *v1.PipelineA
+		var stages = t.collectStages(ctx, pipelineRuns)
+		for _, stage := range stages {
+			podName := stage.name
+			stageName := stage.task
+			if completedStages[stageName] {
 				continue
 			}
-			stageName := pt.task
-			log.Logger().Infof("logging pod: %s for task %s", info(podName), stageName)
+			if stage.podExists {
+				log.Logger().Infof("logging pod: %s for task %s", info(podName), stageName)
+
+				pod, err := t.KubeClient.CoreV1().Pods(t.Namespace).Get(ctx, podName, metav1.GetOptions{})
+				if err != nil && apierrors.IsNotFound(err) {
+					if pa.Spec.Status == v1.ActivityStatusTypeRunning {
+						pa.Spec.Status = v1.ActivityStatusTypeAborted
+					}
+					return nil
+				}
+				if err != nil {
+					return errors.Wrapf(err, "failed to load pod %s in namespace %s", podName, t.Namespace)
+				}
 
-			pod, err := t.KubeClient.CoreV1().Pods(t.Namespace).Get(ctx, podName, metav1.GetOptions{})
-			if err != nil && apierrors.IsNotFound(err) {
-				if pa.Spec.Status == v1.ActivityStatusTypeRunning {
-					pa.Spec.Status = v1.ActivityStatusTypeAborted
+				err = t.getContainerLogsFromPod(ctx, pod, pa, buildName, stageName, out)
+				if err != nil {
+					return errors.Wrapf(err, "failed to get logs for pod %s", podName)
 				}
-				return nil
-			}
-			if err != nil {
-				return errors.Wrapf(err, "failed to load pod %s in namespace %s", podName, t.Namespace)
-			}
 
-			err = t.getContainerLogsFromPod(ctx, pod, pa, buildName, stageName, out)
-			if err != nil {
-				return errors.Wrapf(err, "failed to get logs for pod %s", podName)
-			}
+				// let's reload the pod to see if it's completed or not (it should, now that we've streamed its logs)
+				pod, err = t.KubeClient.CoreV1().Pods(t.Namespace).Get(ctx, podName, metav1.GetOptions{})
+				if err != nil && apierrors.IsNotFound(err) {
+					if pa.Spec.Status == v1.ActivityStatusTypeRunning {
+						pa.Spec.Status = v1.ActivityStatusTypeAborted
+					}
+					return nil
+				}
+				if err != nil {
+					return errors.Wrapf(err, "failed to load pod %s in namespace %s", podName, t.Namespace)
+				}
 
-			// let's reload the pod to see if it's completed or not (it should, now that we've streamed its logs)
-			pod, err = t.KubeClient.CoreV1().Pods(t.Namespace).Get(ctx, podName, metav1.GetOptions{})
-			if err != nil && apierrors.IsNotFound(err) {
-				if pa.Spec.Status == v1.ActivityStatusTypeRunning {
-					pa.Spec.Status = v1.ActivityStatusTypeAborted
+				if pods.IsPodCompleted(pod) {
+					completedStages[stageName] = true
 				}
-				return nil
-			}
-			if err != nil {
-				return errors.Wrapf(err, "failed to load pod %s in namespace %s", podName, t.Namespace)
-			}
-			if pods.IsPodCompleted(pod) {
-				completedPods[podName] = true
+			} else if stage.skipped {
+				completedStages[stageName] = true
+				log.Logger().Infof("pod is skipped/failed for task: %s", stageName)
+			} else {
+				// let's wait second for next pod/task start
+				time.Sleep(time.Second)
```

Comment:
if you want to wait, I think we should do it outside of the inner stages loop, to avoid waiting too much if you have 5 stages for example, and the first one is running (you'll be waiting 4 seconds). I think waiting at the end of the outer loop, if not all stages are marked as completed, should be enough

Reply:
done